deployment/hasura/migrations/Aerie/22_plan_model_migration/down.sql

```diff
@@ -0,0 +1,568 @@
+-- #### PERMISSIONS ####
+
+-- Remove user permissions (see default_user_roles)
+update permissions.user_role_permission
+set function_permissions = function_permissions  - 'migrate_plan_to_model';
+
+-- Need to remove the key 'migrate_plan_to_model' from function_permission_key enum, only way is to drop and recreate it.
+----- 1. drop any functions/procedures/objects that depend on permissions.function_permission_key to avoid errors
+drop function permissions.raise_if_plan_merge_permission(_function permissions.function_permission_key, _permission permissions.permission);
+drop procedure permissions.check_merge_permissions(_function permissions.function_permission_key, _merge_request_id integer, hasura_session json);
+drop procedure permissions.check_merge_permissions(_function permissions.function_permission_key, _permission permissions.permission, _plan_id_receiving integer, _plan_id_supplying integer, _user text);
+drop procedure permissions.check_general_permissions(_function permissions.function_permission_key, _permission permissions.permission, _plan_id integer, _user text);
+drop function permissions.get_function_permissions(_function permissions.function_permission_key, hasura_session json);
+
+------ 2. drop permissions.function_permission_key and recreate without 'migrate_plan_to_model'
+drop type permissions.function_permission_key;
+create type permissions.function_permission_key
+as enum ('apply_preset', 'begin_merge', 'branch_plan', 'cancel_merge', 'commit_merge', 'create_merge_rq',
+  'create_snapshot', 'delete_activity_reanchor', 'delete_activity_reanchor_bulk', 'delete_activity_reanchor_plan',
+  'delete_activity_reanchor_plan_bulk', 'delete_activity_subtree', 'delete_activity_subtree_bulk', 'deny_merge',
+  'get_conflicting_activities', 'get_non_conflicting_activities', 'get_plan_history',  'restore_activity_changelog',
+  'restore_snapshot', 'set_resolution', 'set_resolution_bulk', 'withdraw_merge_rq');
```

Comment:
Nonblocking, Readability: please put the enum entries on separate lines. It makes reviewing the statement easier, as it lets me compare it side by side with how the enum is declared in `permissions.sql`. (Only flagging it as "nonblocking" because I already reviewed this statement).

```suggestion
create type permissions.function_permission_key
as enum (
  'apply_preset',
  'begin_merge',
  'branch_plan', 
  'cancel_merge', 
  'commit_merge', 
  'create_merge_rq',
  'create_snapshot', 
  'delete_activity_reanchor', 
  'delete_activity_reanchor_bulk', 
  'delete_activity_reanchor_plan',
  'delete_activity_reanchor_plan_bulk', 
  'delete_activity_subtree', 
  'delete_activity_subtree_bulk', 
  'deny_merge',
  'get_conflicting_activities', 
  'get_non_conflicting_activities', 
  'get_plan_history',  
  'restore_activity_changelog',
  'restore_snapshot', 
  'set_resolution', 
  'set_resolution_bulk', 
  'withdraw_merge_rq');
```